src/databricks/labs/remorph/reconcile/query_builder/expression_generator.py

```diff
@@ -77,23 +79,28 @@ def anonymous(expr: exp.Expression, func: str) -> exp.Anonymous | exp.Expression
     To achieve the same,we can use the function as below:
     eg:
         >>> expr = parse_one("select col1 from dual")
-        >>> transformed_expr=anonymous(expr,"unix_timestamp")
+        >>> transformed_expr=anonymous(expr,"unix_timestamp({})")
         >>> print(transformed_expr)
         'SELECT UNIX_TIMESTAMP(col1) FROM DUAL'
 
     """
+    level = 0 if isinstance(expr, exp.Column) else 1
     new_expr = expr.copy()
     for node in new_expr.dfs():
         if isinstance(node, exp.Column):
-            return exp.Column(this=func.format(node.name))
+            name = f"{node.table}.{node.name}" if node.table else node.name
+            anonymous_func = exp.Column(this=func.format(name))
+            if level == 0:
+                return anonymous_func
+            node.replace(anonymous_func)
```

Comment:
```suggestion
            if isTerminal:
                return anonymous_func
            node.replace(anonymous_func)
```

Reply:
Done. linting dosent allow isTerminal. So, changes to is_terminal